### PR TITLE
Add `core::convert::absurd`

### PR DESCRIPTION
**Pull request by WaffleLapkin**
r? @Amanieu

I decided to split this from #122412, as this is solely T-libs-api, and can (should?) be dealt with separately from lang/compiler questions.

### CODE REVIEW

**Review by jhpratt — APPROVED** [review]

**Comment by jhpratt** [comment]
LGTM, but not giving the approval as you explicitly requested someone else.

File: library/core/src/convert/mod.rs [hunk 1]
```diff
@@ -105,6 +105,64 @@ pub const fn identity<T>(x: T) -> T {
...
+///
+/// ```
+/// #![feature(never_type, convert_absurd)]
+/// use core::convert::absurd;
+///
+/// let x: Result<_, !> = Ok(1);
+/// let x: u32 = x.unwrap_or_else(absurd);
+/// ```
+///
+/// [`!`]: ../../std/primitive.never.html
```
Comment (lukas-code) [inline library/core/src/convert/mod.rs:158]:
```suggestion
/// [`!`]: prim@never
```

**Comment by Amanieu** [comment]
This was discussed in library team meetings and in person at RustNL and we decided not to move forward with this.

**Pull request closed without merging**